### DISCUSSION TITLE
Full manual garbage collection

### DISCUSSION THREAD

**Discussion by bigpilot** [dotnet/csharplang#7957] [Language Ideas]
Would it be useful to have FULL manual garbage collection in .NET where the program determines when the Garbage Collector runs and to limit the collection time?

new property: 

`GC.Mode = GCCollectionMode.Manual or GCCollectionMode.Automatic   // in addition to the methods that already exist`

new method:

`GC.Collect(int milliseconds)    // only applicable if GC.Mode = Manual`

This would allow time sensitive applications like games or video encoding / decoding engines to be written in a .NET language. 

In addition the GC could also expose some memory pressure statistics to aid the program in determining when the garbage collection should be run.

**Comment by HaloFour** [comment]
I'd suggest opening this suggestion on the [runtime repository](https://github.com/dotnet/runtime) since it doesn't affect the syntax of the language itself.

**Comment by CyrusNajmabadi** [comment]
Not a language request.
